libvcx/src/v3/handlers/connection/connection.rs

```diff
@@ -306,6 +298,10 @@ impl Connection {
      */
     pub fn update_state_with_message(&mut self, message: &A2AMessage) -> VcxResult<()> {
         trace!("Connection: update_state_with_message: {:?}", message);
+        if self.is_in_null_state() {
+            warn!("Connection::update_state_with_message :: update state on connection in null state is ignored");
+            return Ok(());
+        }
```

Comment:
From the POV of the state machine, transitions can still occur even from the null state... Not critical, just a note that this seems like a hack

Reply:
preserved behaviour, this check was previously happening as well, but in difference place. It was taking place in `update_state`. Unlike now, `update_state_with_message `  was previously never called directly outside `connection.rs`

Reply:
I know :) Discussed via IM